### DISCUSSION TITLE
Modbus Communication Error and Driver Issue on BC Roasters BC5

### DISCUSSION THREAD

**Discussion by MeanCatCoffee** [artisan-roaster-scope/artisan#769] [General]
Hello all. Hoping I could get some help with an issue I am having with my Artisan software and my recently acquired, lightly used 2020 BC Roasters BC 5 (6 lb) propane roaster.

I purchased the roaster from someone who had been using Artisan with the roaster with no issues. I'm using the same USB cable that they were using.

Currently using a 2020 MacBook Air with the Intel i3 processor. When I first starting having my issues the laptop had the latest version of Big Sur installed, but since then I have upgraded to the most recent version of Monterey. Issue still persists with the Monterey OS.

I set the up the Artisan software per the instructions on the BC Roasters website and what was listed on the the Artisan website. Installed the CP210x driver per Artisan. Was able to chose this driver when setting up Artisan from the Machine Config menu. First time I used Artisan with the roaster BT and ET displayed properly. Shortly after starting my first roast, Artisan stopped logging roast parameters and displayed "u u" in both the BT and ET fields. Artisan displayed "Modbus Communication Error". I tried several things to get it to work again, including uninstalling and reinstalling both Artisan and the driver. Same issue persisted where it would work for a couple of minutes and the stop working with the communication error.

I next tried installing the CH34x driver because it said some Macs used that. Again, the software would read fine for a while and then get the same error. Sometimes the time log graph would keep moving but the coffee profile graph would stop registering and "u u" would again appear in the ET and BT fields.

Now, after multiple attempts to reinstall the software and drivers, as well as upgrading the OS, there are no driver options shown when setting up the roaster in Artisan. The only choice I have as a driver is COM4 in the Conig-> Machine setup.

I'm hoping someone can help me fix this problem as I would really like to be able to log my roasts and see the data Artisan displays.

**Comment by poundy** [comment]
why are you changing that driver from the CP210X to CH34x driver? Surely only one is correct?  But you seem to have both working at some point, perhaps they're both loaded and you don't know which one the port is actually using?  Upshot - one is unnecessary, one is the correct one and is needed, and only one COM port should be visible to the OS, when the machine is plugged in.

Your symptom, of working fine and then reporting a mid-roast modbus error, to me implies a dodgy USB cable. In the first instance I would suggest replacing that and seeing if the behaviour is corrected.  Modbus uses a relatively simple serial connection, that is time tested, so it's typically something simple that goes wrong.  You might try using the serial log option from the help menu to see what happens, as well as checking messages and errors to see if there are any other important pieces of information in there. 

I have a similar roaster (sold in Australia with different gas trains but essentially the same BlueKing roaster as the BC5 is) and it came with a wireless dongle, but even when using the wired connection I have had no issues like this.

**Comment by MeanCatCoffee** [comment]
I tried the CH34x when the CP210x stopped working. I followed the most recent Artisan setup video from BC Roasters and they said install CP210x driver's and choose that option over COM4 when configuring the machine for Artisan. The CH34X was listed as alternate serial drivers on the Artisan site. Choosing just COM4 results in the Modbus error every time.

I had thought it might be the cable, but like I said, the previous owner used the same cable for many months without issue. I also tried a couple different USB adapters as well (my MacBook has USB C ports instead of standard USB and requires an adapter for normal USB devices. Thanks Apple). I did end up ordering a USB C to printer cable so that I could completely eliminate the use of adapters and have a direct connection between the laptop and roaster, but the cable hasn't arrived yet. 

I had been looking at wireless options. You have more info on the dongle that came with your roaster?

**Reply by poundy** [reply]
the CH34x vs CP210x really is hardware dependent, so Artisan is going to be less specific than your manufacturer. Follow their guidance first (and foremost). As far as which to select in Artisan so you know you're operating against the correct port, I'm afraid I am no help since I am a PC only guy...  Again from a troubleshooting perspective, if something works for a bit then stops, then it's usually not the software you've loaded that's failing, it's the hardware... however that could be in one of four places: the roaster (the USB interface to the modbus network, aka the CP210x module), the cable itself, the USB-C adapter cable, or the mac USB port.  Given the only thing that has changed from when the previous owner was successfully using it, and that USB ports on PC/Mac are pretty reliable, I'd say it's probably your adapter cable that's the bit to swap out first, good to hear you are getting a USB-C to USB-host cable, but sad to hear the few you tried didn't provide better results.

Just to confirm my thinking here, I would remove both CP210x and CH34x drivers from the Mac, and confirm what COM ports can be seen at a system level, to make sure things are as clean as possible. Then, add the CP210x and confirm that once you connect the roaster to the device, a COM port is now available. Then, open Artisan and test that it can open the connection at all - if not, check what port you have selected, and adjust if possible... by doing this and getting so that you can successfully read temperatures from the roast you can confirm that you have the correct logical setup. I'm expecting then that after some time the modbus error will pop up and you'll have some error and serial log data you can share. Can you tell us how far into a roast session it stops ? is it within minutes, or after a few batches, or...? 

My dongle says "HC-06-USB". I have no idea if it's "typical" to have that bluetooth connection in the machine and not give you a dongle, or if the model bought into AUS is configured differently... mine also came with a standard cable.

**Comment by MeanCatCoffee** [comment]
Failure was about 2-3 minutes into roast.

**Comment by MAKOMO** [comment]
Monterey is known to have problems with non-Apple USB-C adapters. Are you using a connector from Apple?

**Reply by MeanCatCoffee** [reply]
No, not Apple. I tried 3 different adapters, 2 of which were Amazon adapters with multiple USB inputs (different brands) and one was an official Google adapter that came with my phone that simply converts USB A to USB C.

**Comment by CarefreeBuzzBuzz** [comment]
I have had no issues with my Anker adapters and one that came with a Western hard drive, nor a multi adapter I use when I roast for a larger monitor and my phidgets.  I think you have to get the basic connection nailed down first so let us know when you have that set up.  Then SAVE your setup please, under the Help menu.   I can’t recall if Brett said this but I am thinking he did, remove both drivers, and then reinstall the one.  Reset Artisan to factory.  Follow the set up and see where it takes you.  

Try running the machine for 30-40 min for warmup and see if anything gets disrupted.  



----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Artisan RoasterScope  <https://artisan-scope.org/>Team

**Comment by MeanCatCoffee** [comment]
Well...good news and bad news. Good news is that Artisan is currently working with my Macbook. Bad news? I have no earthly idea why.

My new cable arrived today, so I was like, "Hey, let's roast some coffee!"

Got roaster started, plugged everything up and...same issue: Modbus communication error. I checked connections reconfigured roaster and nothing. Same error. No BT or ET readings. So I used CleamyMac to completely remove the CP210x drivers, rebooted, reinstalled the CH34x drivers, rebooted, and this time when I configured Artisan I had the option to use the CP210x driver (yes, the drivers I supposedly just uninstalled). So scratching my head I chose the CP210 option instead of COM4 and holy hell Artisan started  working. Went through 4 roasts and it logged everyone of them without issue. I have no idea why the driver I uninstalled showed up and the CH34x I had just installed was not in the list. I bought a Macbook to avoid problems like this...

But I'm not looking a gift horse in the mouth. It's working for now and I'm beyond happy.

Still having small issues with the buttons. I can click Charge and Drop, but as soon as I click on Drop, all the other buttons (First Crack Start, First Crack End, Dry End, etc) gray out and I can't click them. The Cool/End button works fine.

**Reply by poundy** [reply]
FC, FCS and DE are all invalid once you've dropped the coffee !  If you need to add one of those events after you've dropped the coffee, you right click the graph where you want to add, and add....

**Comment by MeanCatCoffee** [comment]
Thanks poundy. You've been a big help.

I checked on Amazon for your bluetooth dongle and it came up with a USB based CP210 bluetooth serial dongle for $17. I can add a female USB A to male USB B adapter for like another $7. I have absolutely no idea if it would work, and I'm kinda scared to monkey around with things at this moment, but for around $25 to add wireless capabilities to my roaster, it may be worth a try.

https://www.amazon.com/dp/B07ZCNL6FT/?coliid=I2WMIPTUK5EHCS&colid=2KNE7789RZCG8&psc=1&ref_=lv_ov_lig_dp_it

https://www.amazon.com/dp/B000AA2SMS/?coliid=I1N5QY528UA6NQ&colid=2KNE7789RZCG8&psc=1&ref_=lv_ov_lig_dp_it

**Reply by poundy** [reply]
ah just to reiterate what my dongle does, as I think you might have misread it in the first instance - my dongle plugs into my PC, and it connects to the existing bluetooth signal generated by the roaster. So for you, the equivalent would need to be a USB-C  to USB-regular cable/connector so you can plug it in your mac, as well as the dongle itself:)  But as I also said, my roaster came with the BT capability and the dongle was already paired/linked with the roaster... so your roaster may already have the capability (I'll take a pic of the control board where I believe the signal is generated, so we can compare - might be a bit until I can get that) or it may not.  In the case of the "not" scenario, something like the above MAY be possible to get working but it's not always as simple as putting another BT module on that end...

**Reply by poundy** [reply]
ok so my interface board has a HC01 module connected to it, doing the outbound communication.  That's paired with the HC06 I connect to the PC.  Here's two pics showing mine that you can compare.
  
![20211220_144023](https://user-images.githubusercontent.com/279966/146709705-681df274-ccb3-493d-86d2-77c11c078aba.jpg)
![20211220_144047](https://user-images.githubusercontent.com/279966/146709722-3c1c73c2-eecd-47a6-952b-9e36243e3dd3.jpg)

**Reply by MeanCatCoffee** [reply]
Ah I see now. I did take a picture of my control board tonight, when I was still having the communication error. I had planned on sending it to BC Roaster's tech support if need be. I was trying to get the lights lighting up to see if it was operating correctly so it's probably not the best angle. I'll get a better shot and compare them, but it almost looks like the board is present on mine as well. We will see.

![PXL_20211219_213114711](https://user-images.githubusercontent.com/96350413/146712478-834e121d-fe5c-4b3f-a4e8-01c44b195597.jpg)

**Reply by poundy** [reply]
that blue LED is on the HC01 board so all good ! 

I haven't tried, but is there a published bluetooth node visible when running, and can you possibly pair to that? Anyway, ask BC-Roasters if thats an option !!

**Comment by Kalebgriffin** [comment]
I’m having a similar problem with my new yoshan roaster. Mine works until I hit the gas ignite button, then that’s when I get the u.u and modbus error. I’ve tried new cables and other drivers but I still can’t get mine to work.

**Reply by poundy** [reply]
unrelated. Lets just talk more about this in your other post
